admin-dev/themes/new-theme/js/pages/module/controller.js

```diff
@@ -0,0 +1,1099 @@
+        const moduleSortingSelector = '.module-sorting-menu';
+        const requiredSelectorCombination = `${moduleGlobalSelector},${moduleSortingSelector}`;
+
+        if (stylesheet.insertRule) {
+          stylesheet.insertRule(
+            requiredSelectorCombination +
+            stylesheetRule, stylesheet.cssRules.length
+          );
+        } else if (stylesheet.addRule) {
+          stylesheet.addRule(
+            requiredSelectorCombination,
+            stylesheetRule,
+            -1
+          );
+        }
+
+        $(self.placeholderGlobalSelector).fadeOut(800, () => {
+          $.each(response.domElements, (index, element) => {
+            $(element.selector).append(element.content);
+          });
+          $(moduleGlobalSelector).fadeIn(800).css('display', 'flex');
+          $(moduleSortingSelector).fadeIn(800);
+          $('[data-toggle="popover"]').popover();
+          self.initCurrentDisplay();
+          self.fetchModulesList();
+        });
+      } else {
+        $(self.placeholderGlobalSelector).fadeOut(800, () => {
+          $(self.placeholderFailureMsgSelector).text(response.msg);
+          $(self.placeholderFailureGlobalSelector).fadeIn(800);
+        });
+      }
+    }).fail((response) => {
+      $(self.placeholderGlobalSelector).fadeOut(800, () => {
+        $(self.placeholderFailureMsgSelector).text(response.statusText);
+        $(self.placeholderFailureGlobalSelector).fadeIn(800);
+      });
+    });
+  }
+
+  fetchModulesList() {
+    const self = this;
+    let container;
+    let $this;
+
+    self.modulesList = [];
+    $('.modules-list').each(function prepareContainer() {
+      container = $(this);
+      container.find('.module-item').each(function prepareModules() {
+        $this = $(this);
+        self.modulesList.push({
+          domObject: $this,
+          id: $this.data('id'),
+          name: $this.data('name').toLowerCase(),
+          scoring: parseFloat($this.data('scoring')),
+          logo: $this.data('logo'),
+          author: $this.data('author').toLowerCase(),
+          version: $this.data('version'),
+          description: $this.data('description').toLowerCase(),
+          techName: $this.data('tech-name').toLowerCase(),
+          childCategories: $this.data('child-categories'),
+          categories: String($this.closest('.modules-list').data('name')).toLowerCase(),
+          type: $this.data('type'),
+          price: parseFloat($this.data('price')),
+          active: parseInt($this.data('active'), 10),
+          access: $this.data('last-access'),
+          display: $this.hasClass('module-item-list') ? self.DISPLAY_LIST : self.DISPLAY_GRID,
+          container,
+        });
+
+        $this.remove();
+      });
+    });
+
+    self.addonsCardGrid = $(this.addonItemGridSelector);
+    self.addonsCardList = $(this.addonItemListSelector);
+    self.updateModuleVisibility();
+    $('body').trigger('moduleCatalogLoaded');
+  }
+
+  updateModuleVisibility() {
```

Comment:
This function is very long which makes it hard to understand

Reply:
Was the old function, just add little condition :/

Reply:
btw I will refactor it